img2dataset/downloader.py

```diff
@@ -19,17 +19,27 @@
 
 
 def download_image(row, timeout):
-    """Download an image with urllib"""
+    """Download an image with requests"""
     key, url = row
     img_stream = None
     try:
-        request = urllib.request.Request(
+        request = requests.get(
             url,
             data=None,
             headers={"User-Agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:72.0) Gecko/20100101 Firefox/72.0"},
+            timeout=timeout,
+            stream=True,
         )
-        with urllib.request.urlopen(request, timeout=timeout) as r:
-            img_stream = io.BytesIO(r.read())
+        if request.status_code != 200:
+            return key, None, "HTTP status code {}".format(request.status_code)
+        start_time = time.time()
+        img_stream = io.BytesIO()
+        for chunk in request.iter_content(chunk_size=1024):
```

Comment:
what if the sender takes 10h to send 1024 bytes?

Reply:
That's right, the timeout from `requests.get` applies both for opening url and receiving packets when reading.
So the slowest is probably `timeout * 1024`.

The issue is a too low `chunk_size` may slow down the process. My test was already a bit slower so I actually think it needs to be increased to `1024 * 1024`.

It's not perfect but can prevent process potentially hanging forever.